src/hevm/src/EVM/UnitTest.hs

```diff
@@ -222,12 +222,18 @@ tick x = Text.putStr x >> hFlush stdout
 
 -- | This is like an unresolved source mapping.
 data OpLocation = OpLocation
-  { srcCode :: ContractCode
+  { srcContract :: Contract
   , srcOpIx :: Int
-  } deriving (Show, Eq, Ord)
+  } deriving (Show)
+
+instance Eq OpLocation where
+  (==) (OpLocation a b) (OpLocation a' b') = b == b' && view contractcode a == view contractcode a'
```

Comment:
Aren't these instances a little weird? A `Contract` contains a lot more than just it's code...

Reply:
well, we only really care about the code here. Previous `OpLocation` was just code and index. We expand it here to be able to better identify contracts, but as far as the insertion to the multiset goes, only the code is relevant

Reply:
I guess the only thing I'm worried about is that at some point in the future someone will use these instances for something else, and encounter some strange and hard to debug issue as a result.

I think I'm probably OK with this for now in the end though, all the additional checks that we would need for a full `Eq` instance seem like they would introduce quite a lot of overhead....